src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif

```diff
@@ -74,42 +82,153 @@
             {
               "analysisTarget": {
-                "uri": "collections/list.cpp",
+                "uri": "file:///home/build/src/collections/list.cpp"
+              },
+              "resultFile": {
+                "uri": "file:///home/build/src/collections/list.h",
                 "region": {
-                  "startLine": 15
+                  "startLine": 15,
+                  "startColumn": 9,
+                  "endLine": 15,
+                  "endColumn": 10,
+                  "length": 1,
+                  "offset": 254
+                }
+              },
+              "fullyQualifiedLogicalName": "collections::list:add",
+              "decoratedName": "?add@list@collections@@QAEXH@Z"
+            }
+          ],
+          "relatedLocations": [
+            {
+              "message": "\"count\" was declared here.",
+              "physicalLocation": {
+                "uri": "file:///home/build/src/collections/list.h",
+                "region": {
+                  "startLine": 8,
+                  "startColumn": 5
                 }
               },
               "fullyQualifiedLogicalName": "collections::list:add"
             }
           ],
           "codeFlows": [
             {
-              "message": "Code flow #1",
+              "message": "Path from declaration to usage",
               "locations": [
                 {
                   "step": 0,
                   "kind": "declaration",
                   "importance": "essential",
-                  "fullyQualifiedLogicalName": "collections::list::add",
+                  "message": "Variable \"p\" declared.",
+                  "snippet": "int *p;",
                   "physicalLocation": {
-                    "uri": "collections/list.cpp",
-                    "uriBaseId": "SRCROOT",
+                    "uri": "file:///home/build/src/collections/list.h",
                     "region": {
-                      "startLine": 12,
-                      "startColumn": 5,
-                      "endLine": 12,
-                      "endColumn": 11,
-                      "offset": 115,
-                      "length": 6
+                      "startLine": 15
                     }
                   },
-                  "snippet": "int n;"
+                  "fullyQualifiedLogicalName": "collections::list:add",
+                  "module": "collections",
+                  "threadId": 52
+                },
+                {
+                  "step": 1,
+                  "kind": "assignment",
+                  "importance": "unimportant",
+                  "snippet": "offset = 0",
+                  "physicalLocation": {
+                    "uri": "file:///home/build/src/collections/list.h",
+                    "region": {
+                      "startLine": 15
+                    }
+                  },
+                  "fullyQualifiedLogicalName": "collections::list:add",
+                  "module": "collections",
```

Comment:
ok, platform.

---
In reply to: [72287911](https://github.com/Microsoft/sarif-sdk/pull/464#discussion_r72287911) [](ancestors = 72287911)